plugins/modules/s3_object.py

```diff
@@ -1281,10 +1289,20 @@ def copy_object_to_bucket(module, s3, bucket, obj, encrypt, metadata, validate,
             if changed:
                 result = {"msg": "tags successfully updated.", "tags": tags}
             return changed, result
-        elif module.check_mode:
-            return True, {"msg": "ETag from source and destination differ"}
-        else:
+        # S3 objects differ
+        if module.check_mode:
             changed = True
+            if s_obj_info.get("ETag") != d_obj_info.get("ETag"):
```

Comment:
If 1 of the src/dest has been uploaded in parts/different parts to the other, the module will always flag a change here, given the comparison of etags. The only non-destructive way to give the correct answer AFAIK is to read to actually checksum the files, which comes with a potentially expensive egress - I suspect this is out of scope for this module?

Reply:
@colin-nolan I have made some updates, the ETag is now computed locally for multipart upload objects to ensure idempotency